src/test/javascript/spec/component/programming-exercise/programming-exercise-reset-dialog.component.spec.ts

```diff
@@ -0,0 +1,234 @@
+import { NgForm, NgModel } from '@angular/forms';
+import { FormDateTimePickerComponent } from 'app/shared/date-time-picker/date-time-picker.component';
+import { FeatureToggleDirective } from 'app/shared/feature-toggle/feature-toggle.directive';
+import { MockHasAnyAuthorityDirective } from '../../helpers/mocks/directive/mock-has-any-authority.directive';
+import { TranslateDirective } from 'app/shared/language/translate.directive';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+
+describe('ProgrammingExerciseResetDialogComponent', () => {
+    let comp: ProgrammingExerciseResetDialogComponent;
+    let fixture: ComponentFixture<ProgrammingExerciseResetDialogComponent>;
+    let programmingExerciseService: ProgrammingExerciseService;
+
+    const exerciseId = 42;
+    const programmingExercise = new ProgrammingExercise(new Course(), undefined);
+    programmingExercise.id = exerciseId;
+    programmingExercise.title = 'Programming Exercise';
+    programmingExercise.releaseDate = dayjs();
+    programmingExercise.dueDate = dayjs().add(7, 'days');
+
+    beforeEach(() => {
+        return TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [
+                ProgrammingExerciseResetDialogComponent,
+                MockTranslateValuesDirective,
+                MockPipe(ArtemisTranslatePipe),
+                MockComponent(FormDateTimePickerComponent),
+                MockDirective(TranslateDirective),
+                MockDirective(FeatureToggleDirective),
+                MockDirective(NgModel),
+                MockDirective(NgForm),
+                MockDirective(MockHasAnyAuthorityDirective),
+            ],
+            providers: [
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                // Ignore console errors
+                console.error = () => false;
+                fixture = TestBed.createComponent(ProgrammingExerciseResetDialogComponent);
+                comp = fixture.componentInstance;
+                programmingExerciseService = fixture.debugElement.injector.get(ProgrammingExerciseService);
+
+                // stubs
+                jest.spyOn(programmingExerciseService, 'find').mockReturnValue(of({ body: programmingExercise } as HttpResponse<ProgrammingExercise>));
+
+                comp.programmingExercise = programmingExercise;
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('test initialization', () => {
+        fixture.detectChanges();
+        expect(comp.programmingExercise.id).toBe(42);
+    });
+
+    it('Programming exercise service should find the correct programming exercise', () => {
```

Comment:
```suggestion
    it('should find the correct programming exercise', () => {
```
I would be nicer if the `it` test description strings would always start with "should"